core/templates/dev/head/components/outcome_destination_editor_directive.html

```diff
@@ -22,7 +22,7 @@
   </div>
   <div style="margin-top: 12px;" ng-if="isSelfLoop() && canAddPrerequisiteSkill">
     <b>Prerequisite skill ID (optional):&nbsp;</b>
-    <input type="text" name="missingPrerequisiteSkillId" ng-model="outcome.missingPrerequisiteSkillId" class="form-control" tabindex="0" aria-invalid="false" ng-pattern="explorationAndSkillIdPattern">
+    <input type="text" name="missingPrerequisiteSkillId" ng-model="outcome.missingPrerequisiteSkillId" class="form-control" tabindex="0" aria-invalid="false" ng-model-options="{allowInvalid: true}" ng-pattern="explorationAndSkillIdPattern">
```

Comment:
Does this mean that a non-empty invalid string can still be saved? If so, that might be a problem.

Reply:
No, it doesn't allow it to be saved. `ng-model-options` just makes sure the scope variable is never undefined, i.e even if it's invalid, the value will be stored in the scope variable, but the Save button will still be disabled and there is still a red outline around the text box (tested it locally). So, I don't think it will be an issue.